### DISCUSSION TITLE
路由规则里geosite更换为rule-set

### DISCUSSION THREAD

**Discussion by toyo2333** [XTLS/Xray-core#3584] [General]
现在使用的v2fly的geosite里既没有gemini的规则，也没有claude （这两个都需要US的IP才能使用），所以没办法在路由的rules里将请求进行分流转发。

我自己去找保护了这两个规则的geo dat文件，不知道去哪里找。 我也不懂得怎么去加。

能不能把这部分做成规则文件，比如clash这样的
https://cdn.jsdelivr.net/gh/blackmatrix7/ios_rule_script@master/rule/Clash/Gemini/Gemini_No_Resolve.yaml
https://cdn.jsdelivr.net/gh/blackmatrix7/ios_rule_script@master/rule/Clash/Claude/Claude_No_Resolve.yaml

这样服务的的路由规则配置也可以更灵活，更新时效性也高一些

**Comment by toyo2333** [comment]
今天修改规则就是觉得麻烦，geodata的更新 还要去提交PR，而现成的rule-set早就实时更新到最小了。

开发者考虑下下不？

**Comment by mmmray** [comment]
https://github.com/v2fly/domain-list-community

**Reply by toyo2333** [reply]
> https://github.com/v2fly/domain-list-community

这里面没有Gemini和Claude的规则，有一个接近的也很旧了。

我的意思是，能不能支持第三方的rule-set ，比如我用clash的，别人及时更新的做好的
https://cdn.jsdelivr.net/gh/blackmatrix7/ios_rule_script@master/rule/Clash/Gemini/Gemini_No_Resolve.yaml
https://cdn.jsdelivr.net/gh/blackmatrix7/ios_rule_script@master/rule/Clash/Claude/Claude_No_Resolve.yaml

**Reply by mmmray** [reply]
I see what you are saying. I thought you had issues finding the source of v2fly to fix it:

> I went to find the geo dat file that protects these two rules myself, but I don’t know where to find it. I don't know how to add it either. 

Those clash rules are very short. I think as a workaround it is easy enough to write direct domain suffix rules into `routing`.

Are you asking to support Clash rules in xray, or to have a way to load multiple rule sets?

**Comment by aseok** [comment]
Hi.
any workaround for gemini.google and ai.google assuming a currently working vpn through zxcvos/Xray-script but failing to correctly handle those with 'something went wrong' gemini.google error?
